tests/test_connections.py

```diff
@@ -242,3 +244,119 @@ def test_post_request_fail():
+    Each "add" and "remove" group should be split into 2 groups each
+    :return:
+    """
+    group_prefix = "G"
+    add_groups = [group_prefix+six.text_type(n+1) for n in range(0, 15)]
+    remove_groups = [group_prefix+six.text_type(n+1) for n in range(15, 30)]
+    user = UserAction(id_type=IdentityTypes.enterpriseID, email="user@example.com")
+    user.add_to_groups(groups=add_groups, group_type=GroupTypes.usergroup) \
+        .remove_from_groups(groups=remove_groups, group_type=GroupTypes.usergroup)
+    assert user.maybe_split_groups(10) is True
+    assert len(user.commands) == 4
+    assert user.commands[0]["add"][GroupTypes.usergroup.name] == add_groups[0:10]
+    assert user.commands[1]["add"][GroupTypes.usergroup.name] == add_groups[10:]
+    assert user.commands[2]["remove"][GroupTypes.usergroup.name] == remove_groups[0:10]
+    assert user.commands[3]["remove"][GroupTypes.usergroup.name] == remove_groups[10:]
+
+
+def test_large_group_action_split():
+    """
+    Ensure that very large group lists (100+) will be handled appropriately
+    Connection.execute_multiple splits commands and splits actions
+    Result should be 2 actions, even though we only created one action
+    :return:
+    """
+    with mock.patch("umapi_client.connection.requests.Session.post") as mock_post:
+        mock_post.return_value = MockResponse(200, {"result": "success"})
+        conn = Connection(**mock_connection_params)
+
+        group_prefix = "G"
+        add_groups = [group_prefix+six.text_type(n+1) for n in range(0, 150)]
+        user = UserAction(id_type=IdentityTypes.enterpriseID, email="user@example.com")
+        user.add_to_groups(groups=add_groups, group_type=GroupTypes.usergroup)
+        assert conn.execute_single(user, immediate=True) == (0, 2, 2)
+
+
+def test_group_size_limit():
+    """
+    Test with different 'throttle_groups' value, which governs the max size of the group list before commands are split
+    :return:
+    """
+    with mock.patch("umapi_client.connection.requests.Session.post") as mock_post:
+        mock_post.return_value = MockResponse(200, {"result": "success"})
+        params = mock_connection_params
+        params['throttle_groups'] = 5
+        conn = Connection(**params)
+
+        group_prefix = "G"
+        add_groups = [group_prefix+six.text_type(n+1) for n in range(0, 150)]
+        user = UserAction(id_type=IdentityTypes.enterpriseID, email="user@example.com")
+        user.add_to_groups(groups=add_groups, group_type=GroupTypes.usergroup)
+        assert conn.execute_single(user, immediate=True) == (0, 3, 3)
+
+
+def test_split_add_user():
+    """
+    Make sure split doesn't do anything when we have a non-add/remove group action
+    :return:
+    """
+    user = UserAction(id_type=IdentityTypes.enterpriseID, email="user@example.com")
+    user.create(first_name="Example", last_name="User", country="US", email="user@example.com")
+    assert user.maybe_split_groups(10) is False
```

Comment:
You also need to check that the length of `user.commands` is still 1.  I think you will find it is 0, due to a bug in maybe_split_groups (noted in the review).  Also check to see that the command has not been altered.  And maybe throw in another non-group command as well, such as `update`.